crates/component/src/wrappers/http.rs

```diff
@@ -0,0 +1,353 @@
+//! This module provides utilities for writing HTTP servers and clients using the WASI HTTP API.
+//!
+//! It's inspired by the WASI 0.3 proposal for <https://github.com/WebAssembly/wasi-http> and will
+//! be supported until the release of wasi:http@0.3.0. After that, this module will be deprecated.
+// Required until <https://github.com/seanmonstar/reqwest/pull/2453> merges
+use reqwest_wasmcloud as reqwest;
+
+use std::{
+    io::{Read, Write},
+    str::FromStr,
+};
+
+use reqwest::header::{HeaderMap, HeaderName, HeaderValue};
+use wasi::{
+    http::{
+        outgoing_handler::ErrorCode,
+        types::{
+            Fields, IncomingBody, InputStream, Method, OutgoingBody, OutgoingResponse,
+            ResponseOutparam, Scheme,
+        },
+    },
+    io::streams::StreamError,
+};
+
+const HTTP_SCHEME: &str = "http";
+const HTTPS_SCHEME: &str = "https";
+
+/// Trait for implementing an HTTP server WebAssembly component that receives a
+/// [`Request`] and returns a [`ResponseBuilder`].
+pub trait HttpServer {
+    fn handle(request: Request) -> Result<ResponseBuilder, ErrorCode>;
+}
+
+pub struct Request {
+    inner: wasi::http::types::IncomingRequest,
+}
+
+impl TryFrom<wasi::http::types::IncomingRequest> for Request {
+    type Error = ErrorCode;
+
+    fn try_from(inner: wasi::http::types::IncomingRequest) -> Result<Self, Self::Error> {
+        Ok(Self { inner })
+    }
+}
```

Comment:
```suggestion
impl From<wasi::http::types::IncomingRequest> for Request {
    fn from(inner: wasi::http::types::IncomingRequest) -> Self {
        Self { inner }
    }
}
```

Reply:
Heh, I saw this one doing another comb through too. Fixing!